numba/tests/test_np_randomgen.py

```diff
@@ -0,0 +1,79 @@
+import numba
+import numpy as np
+
+from numba import types
+from numba.tests.support import TestCase
+from numba.np.random.generator_methods import get_proper_func
+from numpy.random import MT19937, Generator
+from numpy.testing import assert_equal, assert_array_equal, assert_raises
+
+
+def test_proper_func_provider():
+    def test_32bit_func():
+        return 32
+
+    def test_64bit_func():
+        return 64
+
+    assert_equal(get_proper_func(test_32bit_func, test_64bit_func,
+                                 np.float64)[0](), 64)
+    assert_equal(get_proper_func(test_32bit_func, test_64bit_func,
+                                 np.float32)[0](), 32)
+    assert_equal(get_proper_func(test_32bit_func, test_64bit_func,
+                                 types.float64)[0](), 64)
+    assert_equal(get_proper_func(test_32bit_func, test_64bit_func,
+                                 types.float32)[0](), 32)
+
+    # With any other datatype it should return a TypeError
+    with assert_raises(TypeError):
```

Comment:
The Numba test suite "spelling" for this pattern is like this:
https://github.com/numba/numba/blob/4684b9cf9d5e42ec0e144c3f238432b078627a1c/numba/tests/test_np_functions.py#L644-L649
the error type is checked, and then the exception message checked to ensure that it came from the expected place and that it rendered as expected.